### ISSUE TITLE
Improve security-related documentation

### ISSUE THREAD

**Issue by sftim** [kubernetes/website#25119]
**This is an umbrella issue**

**What would you like to be added**
Review and possibly revise the existing pages on the topic of Kubernetes security.

**Why is this needed**
It's important to help readers understand how to make sure a Kubernetes cluster has appropriate security controls.

**Comments**

Look at the overall body of topics related to security context and policy. Do we have the right set of topics? Have we covered all the important points?. Is there overlap that could be reduced? Here are some of the existing topics:
- [Security](https://kubernetes.io/docs/concepts/security/)
- [Controlling Access To The Kubernetes API](https://kubernetes.io/docs/concepts/security/controlling-access/)
- [TLS](https://kubernetes.io/docs/tasks/tls/) and tasks in that section
- [Pod Security Policies](https://kubernetes.io/docs/concepts/policy/pod-security-policy/)
[Securing a Cluster](https://kubernetes.io/docs/tasks/administer-cluster/securing-a-cluster/)
- [Using RBAC Authorization](https://kubernetes.io/docs/reference/access-authn-authz/rbac/)
- [Using Admission Controllers](https://kubernetes.io/docs/admin/admission-controllers/)
- [Network Policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/)
- [Configure a Security Context for a Pod or Container](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/)
After choosing a set of topics, apply the appropriate template: Concept or Task. Edit the topics to comply with our [style guide](https://kubernetes.io/docs/contribute/style/style-guide/). Check for accuracy and also fit against the [CNCF cloud native security white paper](https://github.com/cncf/sig-security/blob/master/security-whitepaper/CNCF_cloud-native-security-whitepaper-Nov2020.pdf)

/sig security

**Comment by kbhawkey** [comment]
The [docs security overview](https://kubernetes.io/docs/concepts/security/overview/#cluster-applications) concept page contains a table of links as well as more links (Secrets, encryption at rest, ...) at the bottom of the page.

**Comment by sftim** [comment]
I reckon it'd be handy if we could find a way to link to https://www.microsoft.com/security/blog/2020/04/02/attack-matrix-kubernetes/

**Comment by sftim** [comment]
Also see issue #27515

**Comment by sftim** [comment]
https://github.com/kubernetes/website/issues/27370#issuecomment-900397304 feels relevant too

**Comment by sftim** [comment]
Is this issue still needed? It might be too broad.

**Comment by shannonxtreme** [comment]
Might be too broad yes. I think there are useful asks in it though. Some have been improved on since this was opened. Maybe this could morph into a collection of the various umbrella issues like ServiceAccount, Secrets, etc? Or we can just close this

**Comment by raesene** [comment]
One point on this is that I *think* "securing a cluster" would be replaced by a combination of the hardening guide that's in progress and the security checklist.

**Comment by sftim** [comment]
Also see https://github.com/kubernetes/website/issues/44169

**Comment by sftim** [comment]
Also see https://github.com/kubernetes/website/issues/44430

**Comment by adityasamant25** [comment]
Will it be good to add more content to the [Security Tutorials](https://kubernetes.io/docs/tutorials/security/) on topics that complement Kubernetes security by using 3rd party tools? I'm thinking of step-by-step guides that people can follow along on their local K8S clusters (minikube?).

Topic examples could be implementing policies using OPA Gatekeeper, Runtime sandboxes like gvisor, trivy for vulnerability scanning etc.
The basic commands and tutorials can be explained with reference to the official documentation of the 3rd party tools if people want to get into the details.

**Comment by kumarankit999** [comment]
Yes, @adityasamant25 I'm agree with you that we can add more content like this above!!

**Comment by sftim** [comment]
> Topic examples could be implementing policies using OPA Gatekeeper, Runtime sandboxes like gvisor, trivy for vulnerability scanning etc. I know these topics are discussed in older blog posts or the community forums, but could be good to have them listed in the official tutorials.

Those might be better handled as newer blog posts. We try not to document too much about 3rd party software, as it is harder to maintain. If the original blog article's author supports it, we could also link a new article to the previous article and vice versa.

**Comment by kumarankit999** [comment]
Yes @sftim, how are we achieving that?

**Comment by sftim** [comment]
Also see https://github.com/kubernetes/website/issues/48546
